internal/handlers/common/compare.go

```diff
@@ -253,3 +257,45 @@ func compareOrdered[T constraints.Ordered](a, b T) compareResult {
 func compareNumbers(a float64, b int64) compareResult {
 	return compareOrdered(a, float64(b))
 }
+
+// matchDocuments returns true if 2 documents are equal.
+func matchDocuments(a, b *types.Document) bool {
+	if a == nil {
+		log.Panicf("%v is nil", a)
+	}
+	if b == nil {
+		log.Panicf("%v is nil", b)
+	}
+
+	keys := a.Keys()
+	if !slices.Equal(keys, b.Keys()) {
+		return false
+	}
+	return reflect.DeepEqual(a.Map(), b.Map())
```

Comment:
no, reflect is less effective, can we use here our `deepEqual`, some code is already written for it.

Reply:
I can't find our `deepEqual`, could you please guide me?

Reply:
it needs to be written

Reply:
got it. I am going to come up with something :)

Reply:
It is less about effectiveness and more about correctness. For example, reflect.DeepEqual doesn’t work like we need it to work for our beloved NaNs.

There is a helper in testutil. Please copy and adopt it; I will refactor those copies later

Reply:
(you can’t use it as-is because testutil depends on testing package, and we don’t want to import in non-test code)

Reply:
OK! Thanks